docs/usage.md

```diff
@@ -0,0 +1,213 @@
+```
+
+___
+
+### Running with Docker
+
+Docker is one quick way for running a vechain node:
+
+```shell
+docker run -d\
+  -v {path-to-your-data-directory}/.org.vechain.thor:/home/thor/.org.vechain.thor\
+  -p 127.0.0.1:8669:8669 -p 11235:11235 -p 11235:11235/udp\
+  --name thor-node vechain/thor --network test
+```
+
+_Do not forget to add the `--api-addr 0.0.0.0:8669` flag if you want other containers and/or hosts to have access to the
+RESTful API. `Thor` binds to `localhost` by default and it will not accept requests outside the container itself without
+the flag._
+
+Release [v2.0.4](https://github.com/vechain/thor/releases/tag/v2.0.4) changed the default user from `root` (UID: 0)
+to `thor` (UID: 1000). Ensure that UID 1000 has `rwx` permissions on the data directory of the docker host. You can do
+that with ACL `sudo setfacl -R -m u:1000:rwx {path-to-your-data-directory}`, or update ownership
+with `sudo chown -R 1000:1000 {path-to-your-data-directory}`.
+
+___
+
+### Docker Compose
+
+A `docker-compose.yml` file is provided for convenience. It will create a container with the same configuration as the
+command above.
+
+```yaml
+version: '3'
+
+services:
+  thor-node:
+    image: vechain/thor
+    container_name: thor-node
+    command: --network test --api-addr 0.0.0.0:8669
+    volumes:
+      - thor-data:/home/thor
+    ports:
+      - "8669:8669"
+      - "11235:11235"
+      - "11235:11235/udp"
+
+volumes:
+  thor-data:
+```
+
+___
+
+### Sub-commands
+
+#### Thor Solo
+
+`thor solo` is a sub-command for running a single node in a standalone mode. It is useful for testing and development.
+
+```shell
+# create new block when there is pending transaction
+bin/thor solo --on-demand
+
+# save blockchain data to disk(default to memory)
+bin/thor solo --persist
+
+# two options can work together
+bin/thor solo --persist --on-demand
+```
+
+#### Master Key
+
+`thor master-key` is a sub-command for managing the node's master key.
+
+```shell
+# print the master address
+bin/thor master-key
+
+# export master key to keystore
+bin/thor master-key --export > keystore.json
+
+
+# import master key from keystore
+cat keystore.json | bin/thor master-key --import
+```
+
+___
+
+### Command line options
+
+To show usages of all command line options:
+
+```shell
+bin/thor -h
+```
+
+| Flag                        | Description                                                                                                |
+|-----------------------------|------------------------------------------------------------------------------------------------------------|
+| `--network`                 | The network to join (main\|test) or path to the genesis file                                               |
+| `--config-dir`              | The directory to use for common configuration files                                                        |
```

Comment:
There is some small diffs in flags: 

```
VERSION:
   2.1.1-88c7c86d-release

COMMANDS:
     solo        client runs in solo mode for test & dev
     master-key  master key management
     help, h     Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --network value              the network to join (main|test) or path to genesis file
   --data-dir value             directory for block-chain databases (default: "/Users/leszek/Library/Application Support/org.vechain.thor")
   --cache value                megabytes of ram allocated to trie nodes cache (default: 4096)
   --beneficiary value          address for block rewards
   --target-gas-limit value     target block gas limit (adaptive if set to 0) (default: 0)
   --api-addr value             API service listening address (default: "localhost:8669")
   --api-cors value             comma separated list of domains from which to accept cross origin requests to API
   --api-timeout value          API request timeout value in milliseconds (default: 10000)
   --api-call-gas-limit value   limit contract call gas (default: 50000000)
   --api-backtrace-limit value  limit the distance between 'position' and best block for subscriptions APIs (default: 1000)
   --api-allow-custom-tracer    allow custom JS tracer to be used tracer API
   --verbosity value            log verbosity (0-9) (default: 3)
   --max-peers value            maximum number of P2P network peers (P2P network disabled if set to 0) (default: 25)
   --p2p-port value             P2P network listening port (default: 11235)
   --nat value                  port mapping mechanism (any|none|upnp|pmp|extip:<IP>) (default: "any")
   --bootnode value             comma separated list of bootnode IDs
   --skip-logs                  skip writing event|transfer logs (/logs API will be disabled)
   --pprof                      turn on go-pprof
   --disable-pruner             disable state pruner to keep all history
   --help, -h                   show help
   --version, -v                print the version
```

Reply:
for e.g.: no config-dir

Reply:
Ya I'm not sure what that is. I took the flags from the codebase

Reply:
^ this is just run of ./bin/thor -h on main :), seems like some flags got hidden field set to true (not sure why)